### ISSUE TITLE
`CA1307` does not respect visibility

### ISSUE THREAD

**Issue by RenderMichael** [dotnet/roslyn-analyzers#6943]
### Analyzer

**Diagnostic ID**: [CA1307](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1307): `Specify StringComparison for clarity`

### Analyzer source
.NET 8
### Describe the bug

``` csharp
F.M(""); // warning here, even though M(string, StringComparison) is private

class F
{
    private static void M(string s, StringComparison c) { }
    public static void M(string s) { }
}
```

### Expected behavior
no warning

### Actual behavior
warning displayed

**Comment by david-acker** [comment]
This is slightly out of the scope of the visibility issue, but how do we want to handle scenarios where an instance method is used but a static `StringComparison` overload exists, and vice versa? I stumbled upon this while writing some tests for the fix.

We only want to report a diagnostic when a `StringComparison` overload exists and it's of the same type (static or instance) as the target method, correct?

Example:

```csharp
public class G : F
{
    public void DoSomething()
    {
        M(""); // We don't want to suggest using F.M(string, StringComparison) here, correct?
    }
}

public class F
{
    public static void M(string s, StringComparison c) { }
    public void M(string s) { }
}
```

**Comment by mavasani** [comment]
> We only want to report a diagnostic when a StringComparison overload exists and it's of the same type (static or instance) as the target method, correct?

That is correct
